packages/driver/src/multi-domain/events/misc.ts

```diff
@@ -0,0 +1,33 @@
+import type { $Cy } from '../../cypress/cy'
+import type { SpecBridgeDomainCommunicator } from '../communicator'
+
+let viewportChangedCallbackFn
+
+export const handleMiscEvents = (Cypress: Cypress.Cypress, cy: $Cy, specBridgeCommunicator: SpecBridgeDomainCommunicator) => {
+  Cypress.on('viewport:changed', (viewport, fn) => {
+    viewportChangedCallbackFn = fn
```

Comment:
Should we create a more robust way of ensuring that the callback you attach is the same callback you call in the viewport changed event? I realize that it might be overkill in this case but we may end up having more events like this where we need to wait for an acknowledgement before we continue. You could assign the event an ID and call the callback with the matching ID or something like that.

Reply:
The short answer is yes, I think a more robust messaging system is probably needed. In this case, Cypress is waiting for the command to finish through the callback function so I don't believe there are any issues but I'll go ahead and log something so we can discuss further.

Reply:
sounds good